lib/add_on.ex

```diff
@@ -44,7 +44,8 @@ defmodule Braintree.AddOn do
 
       {:ok, addons} = Braintree.AddOns.all()
   """
-  @spec all(Keyword.t()) :: {:ok, [t]} | {:error, Error.t()}
+  @spec all(Keyword.t()) ::
+          {:ok, [t]} | {:error, Error.t()} | {:error, atom()} | {:error, binary()}
```

Comment:
I think this (and all of the others) can be written like this instead:

```suggestion
          {:ok, [t]} | HTTP.response()
```

That would remove just about all of this duplication.